connector_clickup/models/project/importer.py

```diff
@@ -0,0 +1,137 @@
+import logging
+from datetime import date, timedelta
+
+from odoo.addons.component.core import Component
+from odoo.addons.connector.components.mapper import mapping, only_create
+
+_logger = logging.getLogger(__name__)
+
+
+class ProjectProjectImporter(Component):
+    _name = "clickup.project.project.importer"
+    _inherit = "clickup.importer"
+    _apply_on = "clickup.project.project"
+
+
+class ProjectProjectBatchImporter(Component):
+    """Delay import of the records"""
+
+    _name = "clickup.project.project.batch.importer"
+    _inherit = "clickup.delayed.batch.importer"
+    _apply_on = "clickup.project.project"
+
+    def run(self, filters=None, force=False):
+        """Run the synchronization"""
+
+        records = self.backend_adapter.search(filters)
+
+        for rec in records:
+            for data in rec.get("lists", []):
+                external_id = data.get(self.backend_adapter._clickup_ext_id_key)
+                self._import_record(
+                    external_id, data=data, force=force, model=self._apply_on
+                )
+
+
+class ProjectProjectImportMapper(Component):
+    _name = "clickup.project.project.import.mapper"
+    _inherit = "clickup.import.mapper"
+    _apply_on = "clickup.project.project"
+    _mapper_ext_key = "identifier"
+
+    # @only_create
+    # @mapping
+    # def odoo_id(self, record):
+    #     """Getting product based on the SKU."""
+    #     print("\n\nodoo_id")
+    #     binder = self.binder_for(model="clickup.project.project")
+    #     project = binder.to_internal(record.get("id"), unwrap=True)
+
+    #     if not project:
+    #         return {}
+
+    #     existing_project = self.env["clickup.project.project"].search(
+    #         [("odoo_id", "=", project.id)], limit=1
+    #     )
+    #     if existing_project:
+    #         return {}
+    #     return {"odoo_id": project.id}
+
+    @only_create
+    @mapping
+    def odoo_id(self, record):
```

Comment:
Only needs to call the method that returns the binding based on external_id, no need for extra logic